modules/thermo-cycler/thermo-cycler-arduino/thermo-cycler-arduino.ino

```diff
@@ -824,15 +826,28 @@ void therm_pid_peltier_update()
 void loop()
 {
   timeStamp = micros();
-  temp_safety_check();
   therm_pid_peltier_update();
+  if (temp_probes.detected_invalid_val)
+  {
+    if (millis() - last_error_print > ERROR_PRINT_INTERVAL)
```

Comment:
If this triggers in or around `millis` overflow time at ~50 days, you're gonna see either
- this firing every single iteration through `loop()` based on underflow because `last_error_print` is near `UNSIGNED_LONG_MAX` and `millis()` is returning a small value
- or it just won't trigger until another 50 days pass

We probably want to do something more like
```c++
unsigned long now = millis();
if (now < last_error_print) {
  last_error_print = now;
}
if (UNSIGNED_LONG_MAX - last_error_print < ERROR_PRINT_INTERVAL) {
   last_error_print = UNSIGNED_LONG_MAX - ERROR_PRINT_INTERVAL;
}
if (now > (last_error_print + ERROR_PRINT_INTERVAL)) {
   gcode.response....
}
```

Reply:
(`UNSIGNED_LONG_MAX` might not be the name of the constant but it or something like it will be in `<limits>` or `<stdint.h>` or something)